### PR TITLE
Make `srclib` library public

### PR DESCRIPTION
**Pull request by jssblck**
# Overview

Adds the currently private `srclib` library, and adds the concept of `organization id` to `Locator`.
This is based on the library here: https://github.com/fossas/foundation/blob/main/lib/srclib/src/lib.rs.

## Acceptance criteria

We can convert to and from locator types.

## Testing plan

Relied on automated tests.

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).

### CODE REVIEW

File: srclib/src/locator.rs [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+/// The intention is not to maintain the list of _all possible fetchers_;
+/// instead the intention is to provide a list of fetchers
+/// with which services in this repository interact.
+///
+/// For this reason this enum is marked non-exhaustive and always will be:
+/// it'll always be possible for fetchers to exist in Core or elsewhere
+/// before they're enumerated here (by choice or by accident).
+///
+/// For more information on the background of `Locator` and fetchers generally,
+/// refer to [Fetchers and Locators](https://go/fetchers-doc).
```
Comment (spatten) [inline srclib/src/locator.rs:345]: This links to a private doc. I think that's probably okay — the link is still useful for people who have access to the doc — but wanted to call it out

**Review by spatten — APPROVED** [review]
:+1:

File: srclib/src/locator.rs (hunk 1 above)
Comment (jssblck) [reply srclib/src/locator.rs:345]: Yeah, that's one reason I left it a golink as well. I want it there for us to reference, but I think we can't make that doc public.

**Review by spatten — APPROVED** [review]
:+1:

**Pull request merged**